group/mod/int.go

```diff
@@ -428,3 +429,34 @@ func reverse(dst, src []byte) []byte {
 	}
 	return dst
 }
+
+func (i *Int) Hash(h kyber.HashFactory, input io.Reader) (kyber.Scalar, error) {
+	canonicalBitLen := i.MarshalSize() * 8
+	actualBitLen := i.M.BitLen()
+	toMask := canonicalBitLen - actualBitLen
```

Comment:
What if this is negative? Could it be?

Reply:
Not with the mod.Int implementation because MarshalSize returns the length in byte and BitLen the length in bits which can not be more than MarshalSize by definition - at least that's the guarantees of big.Int so :man_shrugging: